modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/table/TableCell.tsx

```diff
@@ -142,10 +195,32 @@ function TableCell({
 		);
 	}
 
-	return (
-		<DndTableCell columnName={String(options.fieldName)}>
-			{DataRenderer && !loading ? (
-				<DataRenderer
+	if (!dataRenderer || loading) {
+		return (
+			<DndTableCell columnName={String(options.fieldName)}>
+				<span
+					aria-hidden="true"
+					className="loading-animation loading-animation-sm"
+				/>
+			</DndTableCell>
+		);
+	}
+
+	if (dataRenderer.type === 'clientExtension') {
+		return (
+			<DndTableCell columnName={String(options.fieldName)}>
+				<ClientExtensionRendererComponent
+					args={{value}}
+					renderer={dataRenderer.renderer}
+				/>
+			</DndTableCell>
+		);
+	}
+
+	if (dataRenderer.type === 'internal') {
+		return (
+			<DndTableCell columnName={String(options.fieldName)}>
```

Comment:
This can be an unconditional return, there is no third type.

Reply:
:thinking: Yeah, most probaby. However I'm not sure now if I had to add it so that the TypeScript compiler was happy and acknowledge that the dataRenderer was truly an internal one... I'll check

Reply:
I've removed the `if` and `tsc` doesn't complain :+1: